### PR TITLE
bump Rust

### PR DESCRIPTION
**Pull request by RalfJung**
Mostly to make sure https://github.com/rust-lang/rust/pull/60341 doesn't break us on macOS (Rust CI does not test tools on macOS).

### CODE REVIEW

**Comment by RalfJung** [comment]
Failure only on 32bit Windows... WTF.^^

**Comment by RalfJung** [comment]
I had to add some more special rules to allow `pointer > 1`-style comparisons.

I didn't bother doing this in full generality because hopefully @christianpoveda will have intptrcast finished sooner rather than later, and then we can talk about pretty much requiring that to be enabled to allow executing pre-main stuff...

**Comment by RalfJung** [comment]
@oli-obk does this look good to you?

File: src/operator.rs [hunk 1]
```diff
@@ -109,6 +109,27 @@ impl<'mir, 'tcx> EvalContextExt<'tcx> for super::MiriEvalContext<'mir, 'tcx> {
                     err!(InvalidPointerMath)
                 }
             }
+            Gt | Ge if left.is_ptr() && right.is_bits() => {
```
Comment (oli-obk) [inline src/operator.rs]: I mean it doesn't really matter with int2ptr around the corner, but why is this not implemented commutatively?

**Comment by oli-obk** [comment]
lgtm

File: src/operator.rs (hunk 1 above)
Comment (RalfJung) [reply src/operator.rs]: I didn't find a good way to do this without copying the code and, as you said, with int2ptr around the corner it just does not seem worth the effort.

**Pull request merged**